File: backend/app/nutrition/domain/school_class.py
import re
from dataclasses import dataclass
from uuid import UUID, uuid4

from app.nutrition.domain.mealtime import Mealtime
from app.nutrition.domain.teacher import TeacherID


@dataclass(frozen=True, eq=True)
class ClassID:
    value: UUID

    @classmethod
    def generate(cls) -> "ClassID":
        return cls(uuid4())


@dataclass(frozen=True, eq=True)
class Literal:
    value: str

    _REGEX = re.compile(r"[А-ЯЁ]")

    def __post_init__(self) -> None:
        if not self._REGEX.fullmatch(self.value):
            raise ValueError("Буква класса должны быть заглавной кириллицей")

    def __str__(self) -> str:
        return str(self.value)


@dataclass(frozen=True, eq=True, order=True)
class Number:
    value: int

    def __post_init__(self) -> None:
        if not 1 <= self.value <= 11:
            raise ValueError("Цифра класса должна быть от 1 до 11")

    def __str__(self) -> str:
        return str(self.value)


@dataclass
class SchoolClass:
    id: ClassID
    teacher_id: TeacherID | None
    number: Number
    literal: Literal
    mealtimes: set[Mealtime]

File: frontend/web/src/7_shared/api/implementations/v3/mappers/school-class.ts
import {SchoolClassOut} from '../backend-types/nutrition/school-classes.ts';
import {SchoolClass} from '../frontend-types/nutrition/school-class.ts';
import {toMealtimeArray} from './mealtime.ts';

export const toSchoolClass = (schoolClass: SchoolClassOut): SchoolClass => ({
  id: schoolClass.id,
  teacherId: schoolClass.teacherId,
  number: schoolClass.number,
  literal: schoolClass.literal,
  mealtimes: toMealtimeArray(schoolClass.mealtimes)
});

export const toSchoolClassArray = (classes: SchoolClassOut[]) => classes.map(toSchoolClass);
